dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java

```diff
@@ -0,0 +1,117 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static io.openliberty.checkpoint.fat.FATSuite.getTestMethodNameOnly;
+import static org.junit.Assert.assertTrue;
+
+import org.jboss.shrinkwrap.api.ArchivePath;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import app1.TestServletA;
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.rules.repeater.MicroProfileActions;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class BellsTest extends FATServletClient {
+
+    public static final String APP_NAME = "app1";
+
+    public static final String SERVER_NAME = "checkpointBells";
+
+    @Server(SERVER_NAME)
+    @TestServlet(servlet = TestServletA.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests repeatTest = MicroProfileActions.repeat(SERVER_NAME, TestMode.FULL, //
+                                                                      MicroProfileActions.MP41, // first test in LITE mode
+                                                                      // rest are FULL mode
+                                                                      MicroProfileActions.MP50, MicroProfileActions.MP60);
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        buildAndExportBellLibrary(server, "app1", "AppInitializer");
+        ShrinkHelper.defaultApp(server, APP_NAME, APP_NAME);
+        FATSuite.copyAppsAppToDropins(server, APP_NAME);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        String name = getTestMethodNameOnly(testName);
+        switch (name) {
+            case "testBellsCheckpointAtDeployment":
+                server.setCheckpoint(CheckpointPhase.DEPLOYMENT, false, null);
+                break;
+            case "testBellsCheckpointAtApplication":
+                server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, null);
+                break;
+            default:
+                server.setCheckpoint(CheckpointPhase.APPLICATIONS, true, null);
```

Comment:
I prefer to fail here instead.

Reply:
I am using an existing application app1. It already has a test which requires a server start, that is why I added that logic in default. Please let me know if I should handle it differently. Thanks